src/main/java/org/alfresco/repo/web/scripts/quickshare/QuickShareContentGet.java

```diff
@@ -179,9 +179,24 @@ protected void executeImpl(NodeRef nodeRef, Map<String, String> templateVars, We
         }
         
         // determine attachment
-        boolean attach = Boolean.valueOf(req.getParameter("a"));
-        
+        // MNT-21118 (XSS prevention)
+        // Force the attachment in case of asking for the content file only
+        // (thumbnails don't need this treatment as there not html|xml|...)
+        boolean attach = Boolean.valueOf(req.getParameter("a")) || checkInvokedWebScript(req);
+
         // Stream the content
         streamContentLocal(req, res, nodeRef, attach, propertyQName, model);
 	}
+
+    /**
+     * Check the id of the webscript to know whether it's a quickshare content file invocation
+     * see QuickShareContentGet Bean
+     *
+     * @param req
+     * @return
+     */
+    private boolean checkInvokedWebScript(WebScriptRequest req)

```

Comment:
Why do we need this check?
Is org.alfresco.repo.web.scripts.quickshare.QuickShareContentGet used anywhere else? I could only find one reference to it, in this webscript with id webscript.org.alfresco.repository.quickshare.content-noauth.get

So this check shouls always return true.

Reply:
This method is also used when a document is viewed using:
-  _Share > View_ (_share/s/shared_id_)
- the associated _webscript.org.alfresco.repository.quickshare.content-**thumbnail**-noauth.get_

In this case, since the document is a _pdf_ it is not necessary to force the downloading and the feature of viewing shared documents will no longer be there.